packages/react/src/contents/hooks/useCommercePages.ts

```diff
@@ -1,91 +1,87 @@
- */
-const useCommercePages = (
-  slug: string,
-  params: CommercePagesParams,
-  pageIndex?: number,
-  pageSize?: number,
-  strategy = 'default',
+const useCommercePages = <T = ComponentType[]>(
+  options: UseCommercePageOptions,
 ) => {
-  const queryCommerce = {
-    type: params.type,
-    id: params?.id,
-    gender: params?.gender,
-    brand: params?.brand,
-    category: params?.category,
-    priceType: params?.priceType,
-    sku: params?.sku,
-    pageSize,
-    pageIndex,
-  };
-  const queryContent = {
-    codes: slug,
-    contentTypeCode: 'commerce_pages',
-    pageIndex,
-    pageSize,
-  };
+  const {
+    enableAutoFetch = true,
+    fetchConfig,
+    strategy = Strategy.Default,
+    ...fetchQuery
+  } = options;
+
+  const query = useMemo(
+    () => ({
+      contentTypeCode: ContentTypeCode.CommercePages,
+      ...fetchQuery,
+    }),
+    [fetchQuery],
+  );
   const error = useSelector((state: StoreState) =>
-    getContentError(state, queryContent),
+    getContentError(state, query),
   );
   const isLoading = useSelector((state: StoreState) =>
-    isContentLoading(state, queryContent),
+    isContentLoading(state, query),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isContentFetched(state, query),
   );
-  const commercePage = useSelector((state: StoreState) =>
-    getContents(state, queryContent),
+  const entries = useSelector((state: StoreState) =>
+    getContents<T>(state, query),
+  );
+  const contentGroup = useSelector((state: StoreState) =>
+    getContentByQuery(state, query),
+  );
+  const fetchCommercePagesAction = useAction(fetchCommercePages);
+
+  const fetch = useCallback(
+    () => fetchCommercePagesAction(fetchQuery, strategy, fetchConfig),
+    [fetchCommercePagesAction, fetchQuery, strategy, fetchConfig],
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: contentGroup?.number,
+      totalPages: contentGroup?.totalPages,
+      totalItems: contentGroup?.totalItems,
+    }),
+    [contentGroup],
   );
-  const action = useAction(fetchCommercePagesAction);
-  const fetchCommercePages = () => action(queryCommerce, slug, strategy);
-  const resetContents = useAction(resetContentsAction);
 
   useEffect(() => {
-    !commercePage && fetchCommercePages();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [commercePage, slug]);
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
+      fetch();
+    }
+  }, [enableAutoFetch, error, fetch, isFetched, isLoading]);
 
   return {
-    /**
-     * Get the result for a specific Commerce Page.
-     */
-    commercePage,
-    /**
-     * Reset contents.
-     */
-    resetContents,
-    /**
-     * Loading state for a specific Commerce Page.
-     */
     isLoading,
-    /**
-     * Error state for a specific Commerce Page.
-     */
+    isFetched,
     error,
-    /**
-     * Fetch Commerce Page content for a specific slug.
-     */
-    fetchCommercePages,
+    data: {
```

Comment:
If `entries` is undefined, `data` will contain an object with all values undefined. I think it would be best if you changed the data calculation to a `useMemo` where you would check if `entries` is undefined, you would return `undefined`. Another question: Can `entries` be undefined but `pagination` not be undefined? If that is not possible, then I would change the `data` calculation to be something like this:

```
const data = useMemo(() => {
  if(!entries || !contentGroup) {
     return undefined;
  }

  return {
   entries:
   pagination: {
      pageIndex: contentGroup?.number,
      totalPages: contentGroup?.totalPages,
      totalItems: contentGroup?.totalItems,
   }

}, [entries, contentGroup]);
```